### PR TITLE
MAGN-7835 Allow users to request customizer invitation

### PR DESCRIPTION
**Pull request by ramramps**
### Purpose

This PR has the changes that allows users to request customizer invitation
This PR has only Dynamo Changes. 
Flood changes are tracked in PR : https://github.com/DynamoDS/Flood/pull/84
A new menu item (Sign up for Customizer…) is created in Dynamo. This is enabled only when there is an AuthProvider. This menu item opens a new dialog which connects to the flood server. Email request will be sent from flood. 

Link to YouTrack:
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7835
### Declarations

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@pboyer

### CODE REVIEW

File: src/DynamoPublish/Resource.resx [hunk 1]
```diff
@@ -189,4 +189,25 @@
...
+     <value>Request Invite</value>
+  </data>
+  <data name="InviteRequestStart" xml:space="preserve">
+    <value>Sending Request...</value>
+  </data>
+  <data name="InviteRequestSuccess" xml:space="preserve">
+    <value>Request sent successfully.</value>
+  </data>
+  <data name="InviteRequestFailed" xml:space="preserve">
+    <value>Failed to send a request. Try again later</value>
```
Comment (pboyer) [inline src/DynamoPublish/Resource.resx]: Failed to send the request. Try again later.

**Comment by pboyer** [comment]
@ramramps Awesome! Just a tiny request and this should be good.

**Comment by ramramps** [comment]
@pboyer - PTAL

**Comment by pboyer** [comment]
@ramramps This doesn't build!
```
"C:\projects\dynamo\src\Dynamo.All.2013.sln" (default target) (1) ->
"C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj" (default target) (78) ->
(CoreCompile target) -> 
  Models\InviteModel.cs(64,37): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(68,37): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(72,37): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(90,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(106,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(112,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(130,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  Models\InviteModel.cs(133,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  DynamoPublishExtension.cs(116,27): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  ViewModels\InviteViewModel.cs(98,39): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
  ViewModels\InviteViewModel.cs(106,43): error CS0103: The name 'Resource' does not exist in the current context [C:\projects\dynamo\src\DynamoPublish\DynamoPublish.csproj]
 
    917 Warning(s)
    11 Error(s)
```

**Comment by ramramps** [comment]
@pboyer - Fixed the build. PTAL.

File: src/DynamoPublish/Models/InviteModel.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                return false;
+            }
+
+            if (authenticationProvider == null)
+            {
+                OnUpdateStatusMessage(Resources.AuthenticationFailedMessage, true);
+                return false;
+            }
+
+            string fullServerAdress = serverUrl + ":" + port;
```
Comment (pboyer) [inline src/DynamoPublish/Models/InviteModel.cs]: @aosyatnik Recently merged in a PR that removes the port field.

File: src/DynamoPublish/Models/InviteModel.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            if (restClient == null)
+            {
+                restClient = new RestClient(fullServerAdress);
+            }
+            
+            var  request = new RestRequest(invite, Method.POST);
+            
+            authenticationProvider.SignRequest(ref request,restClient);
+        
+            var response = restClient.Execute(request);
```
Comment (pboyer) [inline src/DynamoPublish/Models/InviteModel.cs]: Are you sure none of these could throw an exception?

File: src/DynamoPublish/Models/InviteModel.cs [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            // Open the configuration file using the dll location.
+            var config = ConfigurationManager.OpenExeConfiguration(this.GetType().Assembly.Location);
+            // Get the appSettings section.
+            var appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
+            serverUrl = appSettings.Settings["ServerUrl"].Value;
+            if (String.IsNullOrWhiteSpace(serverUrl))
+                throw new Exception(Resources.ServerNotFoundMessage);
+
+            port = appSettings.Settings["Port"].Value;
```
Comment (pboyer) [inline src/DynamoPublish/Models/InviteModel.cs]: DynamoPublish.dll.config no longer contains a "Port" field.

**Comment by ramramps** [comment]
@pboyer - Removed the port. Added a try catch for sign in request.

**Comment by pboyer** [comment]
LGTM. Let's fix the merge conflict, though.

**Pull request merged**